src/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  selectedPerson: Person | null;
+  onSelect?: (person: Person) => void;
+  delay: number
+};
+
+export const DropDown: React.FC<Props> = React.memo(({
+  people,
+  selectedPerson,
+  onSelect = () => {},
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(debounce(setApliedQuery, delay), []);
+
+  const visiblePeople = useMemo(() => (people.filter(person => {
+    const normalizedQuery = apliedQuery.trim().toLowerCase();
+
+    return person.name.trim().toLowerCase().includes(normalizedQuery);
+  })), [apliedQuery, people]);
+
+  const resultToShow = (visiblePeople.length > 0);
+
+  const isPeopleListShown = !selectedPerson
+  && resultToShow
+  && isFocused;
+
```

Comment:
I don't understand why do you need so many checks? You can just show no matching message or visiblePeople (depending on the list length) if input is focused
Maybe because of thet it's not working correctly
After i seacrh non-existing person ![image](https://github.com/mate-academy/react_autocomplete/assets/44444911/651b0560-e192-47a3-a9c8-08655b793342)
It's not showing the list if I erased input 
![image](https://github.com/mate-academy/react_autocomplete/assets/44444911/0c1d88ec-57f3-4b1d-a8b5-a0db43e1b274)

Reply:
Actually filtering works correctly, if enter some invalid input and than delete it: https://gyazo.com/87be1429ea480fe7475372f6820cb403

I add addtitional check to hide list if person is selected. Other checks are nessesary - to check if filtered list has length and if input is focused. I create variables for it to make code more clear.
I don't use conditional operator in this case, because if filtered list is empty, so message 'No matching suggestions' to be shown, it will be shown like list element - on list background. And it doesn't look good, as for me it should be just paragraph under input.